### ISSUE TITLE
markdownlint.config is not applied on remote SSH workspaces

### ISSUE THREAD

**Issue by m5ls7h** [DavidAnson/vscode-markdownlint#234]
Working on workspaces with remote SSH, `markdownlint.config` entries are detected on the remote workspace, but are ignored in linting.
If I change the setting in a configuration file on the remote machine, markdownlint output shows
```console
[10:26:23 AM] INFO: Resetting configuration cache due to setting change.
[10:26:23 AM] INFO: Resetting ignore cache due to setting change.
```
but the setting is ignored.
Example setting:
```json
"markdownlint.config": {
	"first-line-heading": { "level": 2},
}
```
On the remote server, I tried to add the setting to the `.workspace` file, to `.vscode\settings.json`, and to `.markdownlint.json` files. All are detected in the markdownlint output but are ignored by linting.
Linting in general works on remote SSH workspaces, but only with the default rule settings.

The same  `markdownlint.config`  setting for a workspace on my local machine is accepted.

**Comment by DavidAnson** [comment]
If this is a virtual workspace scenario in VS Code (and I think it may be), you will see a warning like "In web scenarios, access to configuration files is not supported. Default linting settings are used instead." associated with the extension. (Look for a warning triangle or similar in the extension tab on the left.) If so, this behavior is expected for now. If not, please let me know as I'll have to figure out how to try it myself.

Another way to check is to run the markdownlint.lintWorkspace command/task and look for the error "No workspace folder or Node modules unavailable.".

**Comment by DavidAnson** [comment]
The option to lint the workspace is only displayed if a workspace (folder) is open which suggests that is not the case here. However, watching for edits to configuration files only happens when a workspace is open which suggests that is the case here. I'm not really sure what's going on in your scenario and may need to try myself. What version of the markdownlint extension is being used and are you using an extension for SSH access?

**Comment by m5ls7h** [comment]
I further investigated this issue and found the problem.
My workspace includes two files
- docs/.markdownlint-cli2.jsonc
- docs/.markdownlint.jsonc
If these files are present, any additional settings in the workspace, any markdownlint setting in the `.code-workspace` settings file are detected (as indicated by the markdownlint output) but ignored.

In <https://github.com/DavidAnson/markdownlint-cli2#configuration> there is a precedence of files mentioned, but no indication that vscode settings are ignored if these files are present.

Is this an intendend behavior with documentation missing (or maybe not found by me in the documentation)?

**Comment by m5ls7h** [comment]
My environment:

- markdownlint v0.46.0
- vscode Version: 1.65.0 (user setup)

**Comment by DavidAnson** [comment]
CLI2 does not "know" about VS Code and so does not mention that. The VS Code extension explains its configuration precedence at the end of this section: https://github.com/DavidAnson/vscode-markdownlint#markdownlintconfig

It sounds like that explains what you are seeing?
